arq/main.py

```diff
@@ -120,18 +120,22 @@ def _bind_decorators(self):
         if self._concurrency_enabled:
             redis = self.redis or await self.get_redis()
             main_logger.debug('%s.%s → %s', self.name, func_name, queue)
-            await self.job_future(redis, queue, func_name, *args, **kwargs)
+            j = await self.job_future(redis, queue, func_name, *args, **kwargs)
         else:
             main_logger.debug('%s.%s → %s (called directly)', self.name, func_name, queue)
             data = self.job_class.encode(class_name=self.name, func_name=func_name, args=args, kwargs=kwargs)
             j = self.job_class(data, queue_name=queue)
             await getattr(self, j.func_name).direct(*j.args, **j.kwargs)
+        return j
 
-    def job_future(self, redis, queue: str, func_name: str, *args, **kwargs):
-        return redis.rpush(
+    async def job_future(self, redis, queue: str, func_name: str, *args, **kwargs):
+        data = self.job_class.encode(class_name=self.name, func_name=func_name, args=args, kwargs=kwargs)
```

Comment:
I don't like encoding then decoding here.

Better to modify `Job` to something like

```py
class Job:
    def __init__(self, *, job_id: str=None, queued_at: int=None, ...):
       self.id = self.generate_id(job_id)
       ...

    def encode(self):
        ...
    
    def decode(cls, raw_data: bytes, *, queue_name: str=None, raw_queue: bytes=None): -> Job:
        # like current __init__ method
```

Reply:
I add this functionality in #92